lib/elixir/lib/exception.ex

```diff
@@ -1107,37 +1107,46 @@ defmodule FunctionClauseError do
 
     mfa = Exception.format_mfa(module, function, arity)
 
-    formatted_args =
-      args
-      |> Enum.with_index(1)
-      |> Enum.map(fn {arg, i} ->
-        ["\n    # ", Integer.to_string(i), "\n    ", pad(inspect_fun.(arg)), "\n"]
-      end)
-
-    formatted_clauses =
-      if clauses do
-        format_clause_fun = fn {args, guards} ->
-          code = Enum.reduce(guards, {function, [], args}, &{:when, [], [&2, &1]})
-          "    #{kind} " <> Macro.to_string(code, ast_fun) <> "\n"
-        end
+    format_clause_fun = fn {args, guards} ->
+      code = Enum.reduce(guards, {function, [], args}, &{:when, [], [&2, &1]})
+      "    #{kind} " <> Macro.to_string(code, ast_fun) <> "\n"
+    end
 
-        top_10 =
-          clauses
-          |> Enum.take(10)
-          |> Enum.map(format_clause_fun)
+    "\n\nThe following arguments were given to #{mfa}:\n" <>
+      "#{format_args(args, inspect_fun)}" <>
+      "#{format_clauses(clauses, format_clause_fun)}"
+  end
 
-        [
-          "\nAttempted function clauses (showing #{length(top_10)} out of #{length(clauses)}):",
-          "\n\n",
-          top_10
-        ]
-      else
-        ""
-      end
+  defp format_args(args, inspect_fun) do
+    args
+    |> Enum.with_index(1)
+    |> Enum.map(fn {arg, i} ->
+      [pad("\n# "), Integer.to_string(i), pad("\n"), pad(inspect_fun.(arg)), "\n"]
+    end)
+  end
+
+  defp format_clauses(clauses, format_clause_fun, limit \\ 10)
```

Comment:
I am not the biggest fun of default arguments for private functions but I understan the concern is to remove the magic number. What do you think about having a module attribute called `@clauses_limit` and passing it explicitly instead?

Everything else is perfect, thank you!

Reply:
Makes sense was also considering a module attribute. Have updated